pkg/chains/storage/grafeas/grafeas_test.go

```diff
@@ -0,0 +1,415 @@
+/*
+Copyright 2020 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grafeas
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/chains/pkg/chains/formats"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
+	attestationpb "github.com/grafeas/grafeas/proto/v1beta1/attestation_go_proto"
+	commonpb "github.com/grafeas/grafeas/proto/v1beta1/common_go_proto"
+	pb "github.com/grafeas/grafeas/proto/v1beta1/grafeas_go_proto"
+	"github.com/tektoncd/chains/pkg/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	gstatus "google.golang.org/grpc/status"
+	"google.golang.org/protobuf/testing/protocmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	logtesting "knative.dev/pkg/logging/testing"
+)
+
+type args struct {
+	tr        *v1beta1.TaskRun
+	payload   []byte
+	signature string
+	opts      config.StorageOpts
+}
+
+type testConfig struct {
+	name    string
+	args    args
+	wantErr bool
+}
+
+type mockGrafeasV1Beta1Server struct {
+	// Embed for forward compatibility.
+	// Tests will keep working if more methods are added in the future.
+	pb.UnimplementedGrafeasV1Beta1Server
+
+	// Assume there is only one project for storing notes and occurences
+	occurences map[string]*pb.Occurrence
+	notes      map[string]*pb.Note
+}
+
+var grafeasServer mockGrafeasV1Beta1Server
```

Comment:
This works for now since we only have 1 test, but if we ever have multiple tests run in parallel this single package level var will cause contention. I'd remove this and just create new instances in `setupConnection` instead.